packages/design-system/src/components/popup/popup.tsx

```diff
@@ -85,50 +110,62 @@ function Popup({
             ignoreMaxOffsetY,
             offsetOverride,
           })
-        : {};
-      const { height, x, width, right } = popup.current
-        ? popup.current.getBoundingClientRect()
-        : {};
+        : EMPTY_OFFSET;
+      const popupRect = popup.current?.getBoundingClientRect();
 
-      const updatedXOffset = () => {
+      const updatedXOffset = (): Offset | null => {
+        if (!popupRect) {
+          return null;
+        }
```

Comment:
Damn, this bug took me all day to find. Basically, the old code destructured an empty object into DOMRect attributes in case of the popup not showing yet and depended on `undefined < someLimit === NaN`, which is falsey. I used a default value of 0 in the destructuring to make TS happy, but that actually made the statement true, because suddenly it _was_ smaller than the limit value, which `undefined` _"isn't"_. So with the proper checks for the missing DOMRect just skipping this whole section of code, and everything works again.